packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -550,10 +562,12 @@ export class MirrorNodeClient {
             }
         }
 
-        return {
+        const response = {
             type,
             entity: data.value
         };
+        this.cache.set(cachedLabel, response);
```

Comment:
maybe we should only cache on accounts that exist? Let's see

Reply:
If we look at the entire function, we will see that there are 3 possibilities:

1. The function receives a valid contract address
    - We query for the contract address, if the contract exists, then return and cache it. The tricky part here is that we MUST check for the contract existence first because the mirror node should return a valid response for both `/contracts/<contractAddress>` and `/accounts/<contractAddress>` because in hedera a contract is treated the same way as an account entity

2. The function receives a valid account or token (we run them with `Promise.any` which will fail only when the 2 promises have been rejected)
    - The account or token has been found, we return it to the user and cache it

3. The function receives a non-existing contract/account/token
    - The contract and token endpoints return null - no entity has been found we return null from the function and we do not cache the response
    - The account and token promises were rejected because of `Promise.any` - no entity has been found, do not cache

Reply:
Gotcha.
Thanks for the reminders and explanation.
I think the fact that the whole function is hard to follow to this point is something to be address.
Either with future refactoring or with a few more comments in the function